edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/Serializer/DeflateSerializer.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.MetricsCollector
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.IO.Compression;
+    using System.Text;
+
+    /// <summary>
+    /// Simple wrapper for the System.IO.Compression.DeflateStream class to make it work on byte arrays.
+    /// </summary>
+    public static class DeflateSerializer
+    {
+        public static byte[] Compress(IEnumerable<byte> bytes)
```

Comment:
Why not use Gzip? We already have helpers for that.

Reply:
Didn't know the Gzip helper existed. I'll switch to it